src/components/ProfilePage/QuestCarousel/QuestCarousel.tsx

```diff
@@ -1,89 +1,104 @@
+                let completed = false;
+                if (rewardsIds && pastCampaigns) {
+                  completed = checkInclusion(pastCampaigns, rewardsIds);
+                }
 
-                  //todo: exclude in a dedicated helper function
-                  let completed = false;
-                  if (rewardsIds && pastCampaigns) {
-                    completed = checkInclusion(pastCampaigns, rewardsIds);
-                  }
+                return (
+                  <QuestCardDetailled
+                    key={`available-mission-${index}`}
+                    active={true}
+                    title={quest?.attributes.Title}
+                    image={String(imgURL)}
+                    points={quest?.attributes.Points}
+                    link={quest?.attributes.Link}
+                    startDate={quest?.attributes.StartDate}
+                    endDate={quest?.attributes.EndDate}
+                    platformName={
+                      quest?.attributes.quests_platform?.data?.attributes?.Name
+                    }
+                    slug={quest?.attributes.Slug}
+                    chains={chains}
+                    rewards={rewards}
+                    completed={completed}
+                    claimingIds={claimingIds}
+                    variableWeeklyAPY={
+                      quest?.attributes.Points > 0 && rewardType === 'weekly'
+                    }
+                    rewardRange={rewardRange}
+                  />
+                );
+              })}
 
-                  return (
-                    <QuestCardDetailled
-                      key={`available-mission-${index}`}
-                      active={true}
-                      title={quest?.attributes.Title}
-                      image={String(imgURL)}
-                      points={quest?.attributes.Points}
-                      link={quest?.attributes.Link}
-                      startDate={quest?.attributes.StartDate}
-                      endDate={quest?.attributes.EndDate}
-                      platformName={
-                        quest?.attributes.quests_platform?.data?.attributes
-                          ?.Name
-                      }
-                      slug={quest?.attributes.Slug}
-                      chains={chains}
-                      rewards={rewards}
-                      completed={completed}
-                      claimingIds={claimingIds}
-                      variableWeeklyAPY={
-                        quest?.attributes.Points > 0 && rewardType === 'weekly'
-                      }
-                      rewardRange={rewardRange}
-                    />
-                  );
-                })
-              : Array.from({ length: 3 }, () => 42).map((_, idx) => (
-                  <QuestCardSkeleton key={'mission-card-skeleton-' + idx} />
+              {/** render ongoing numeric quests */}
+              {ongoingNumericQuests &&
```

Comment:
what do you think about extracting this in its own component @dennyscode @oktapodia ?